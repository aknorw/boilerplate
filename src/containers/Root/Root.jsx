import React from 'react'
import PropTypes from 'prop-types'
import { Helmet } from 'react-helmet'
import { compose } from 'redux'
import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import { selectFoo } from 'services/basic/selectors'

import routes from 'pages'

import { RootWrapper, SwitchWrapper } from './components/Layout'
import Navbar from './components/Navbar'

export const Root = ({ foo }) => (
  <RootWrapper>
    <Helmet
      titleTemplate={`%s - ${APP_NAME}`}
      defaultTitle={APP_NAME}
    >
      {/* The following line will be used when implementing i18n */}
      <html lang="en" />
    </Helmet>
    <Navbar title={foo} />
    <SwitchWrapper>
      {routes}
    </SwitchWrapper>
  </RootWrapper>
)

Root.propTypes = {
  foo: PropTypes.string.isRequired,
}

const mapStateToProps = createStructuredSelector({
  foo: selectFoo(),
})

const withConnect = connect(mapStateToProps, null)

export default compose(
  withRouter,
  withConnect,
)(Root)
