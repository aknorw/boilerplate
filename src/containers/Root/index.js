// Make App hot-reloadable in development
// We can include it in build because of the low footprint
import { hot } from 'react-hot-loader'

import ConnectedRoot from './Root'

export default hot(module)(ConnectedRoot)
