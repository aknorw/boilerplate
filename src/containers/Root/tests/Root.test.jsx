import React from 'react'
import { shallow } from 'enzyme'

import { Root } from '../Root'

console.error = jest.fn()

describe('<Root />', () => {
  it('should be defined', () => {
    expect(Root).toBeDefined()
  })
  it('should log 2 errors when no props are passed', () => { // 1 for Root, 1 for Navbar
    shallow(<Root />)
    expect(console.error).toHaveBeenCalledTimes(2)
  })
  it('should render correctly', () => {
    const foo = 'bar'
    const tree = shallow(
      <Root
        foo={foo}
      />,
    )
    expect(tree).toMatchSnapshot()
  })
})
