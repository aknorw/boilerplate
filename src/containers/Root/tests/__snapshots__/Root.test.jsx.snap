// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> should render correctly 1`] = `
<styled.div>
  <HelmetWrapper
    defaultTitle="JEST"
    defer={true}
    encodeSpecialCharacters={true}
    titleTemplate="%s - JEST"
  >
    <html
      lang="en"
    />
  </HelmetWrapper>
  <Navbar
    title="bar"
  />
  <SwitchWrapper>
    <Route
      component={[Function]}
      key="/settings"
      path="/settings"
    />
    <Route
      component={[Function]}
      key="/"
      path="/"
    />
  </SwitchWrapper>
</styled.div>
`;
